packages/loot-core/src/platform/client/fetch/index.web.ts

```diff
@@ -141,3 +137,18 @@ export const listen: T.Listen = function (name, cb) {
 export const unlisten: T.Unlisten = function (name) {
   listeners.set(name, []);
 };
+
+async function closeSocket(onClose) {
+  socketClient.onclose = event => {
+    socketClient = null;
+    onClose();
+  };
+
+  await socketClient.close();
+}
+
+export const clearServer: T.ClearServer = async function () {
+  if (socketClient != null) {
+    return new Promise(resolve => closeSocket(resolve));
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
    return new Promise(closeSocket);
```

Reply:
how do I handle closeSocket's parameter here? I am waiting for the sockets `onclose` event to fire before i return?

Reply:
Both of these snippets of code are identical. One is just shorter.

```
new Promise(resolve => closeSocket(resolve));
new Promise(closeSocket);
```